library/src/lib/scroll-spy/scroll-spy.directive.ts

```diff
@@ -1,26 +1,46 @@
 import { Directive, ElementRef, EventEmitter, HostListener, Input, Output } from '@angular/core';
 
+/**
+ * A directive designed to help navigation elements determine the element currently in view of the user. 
+ */
 @Directive({
     selector: '[fdScrollSpy]'
 })
 export class ScrollSpyDirective {
-
+    /**
+     * An array of tags to track.
+     */
     @Input()
     public trackedTags: string[] = [];
 
+    /**
+     * Whether an arrow is displayed when the user scrolls to the absolute top or bottom.
+     * To hide the arrow, set the property to *true*.
+     */
     @Input()
     public fireEmpty: boolean = false;
 
+    /** 
+     * A number that represent at what location in the container the event is fired. 
+     * 0.5 would fire the events in the middle of the container, 
+     * 0 for the top and 1 for the bottom.
+     */
     @Input()
     public targetPercent: number = 0;
 
+    /** 
+     * Event fired on the scroll element when a new item becomes activated by the scrollspy . 
+     */
     @Output()
     public spyChange: EventEmitter<HTMLElement> = new EventEmitter<HTMLElement>();
```

Comment:
add readonly pls.

also maybe say that the returned value (through $event) is the HTMLElement itself